src/Workflow.cc

```diff
@@ -38,7 +38,7 @@ Workflow::Workflow( const WorkflowStepList & steps )
 
     checkDuplicateIds();
 
-    foreach ( WorkflowStep * step, _steps )
+    for ( WorkflowStep * step : std::as_const( _steps ) )
```

Comment:
Why do we need "syntactic bitter pills" (the opposite to "syntactic sugar") here? This makes the code worse to read. Is there any advantage to offset this?

Even if this might save a handful of CPU cycles (which I am not sure about), those loops never run for more than a handful (< 10) iterations. We would be trading code readability for a miniscule performance improvement, if there even is any.